components/board/genericlinux/data.go

```diff
@@ -96,16 +97,22 @@ func GetGPIOBoardMappings(modelName string, boardInfoMappings map[string]BoardIn
 // getCompatiblePinDefs returns a list of pin definitions, from the first BoardInformation struct
 // that appears compatible with the machine we're running on.
 func getCompatiblePinDefs(modelName string, boardInfoMappings map[string]BoardInformation) ([]PinDefinition, error) {
-	const compatiblePath = "/proc/device-tree/compatible"
+	var path string
+	const (
+		compatiblePath    = "/proc/device-tree/compatible"
```

Comment:
[minor] This sounds like it's for all boards, but you've implemented the rest as though this is only for device-tree boards. I'd put a `Dt` suffix on the name, similar to the `Dmi` suffix on the next line.